lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -295,7 +307,9 @@ SmallVector<Value> LayoutPropagation::propagateToUsers(Value value,
     }
     if (user->hasTrait<mlir::OpTrait::SameOperandsAndResultEncoding>() ||
         user->hasTrait<mlir::OpTrait::Elementwise>() ||
-        isa<triton::ReduceOp, triton::ExpandDimsOp,
+        // TODO(jlebar): Add TransOp to this list, but we have to force it into
+        // shared memory if it's used by a dot operand.
```

Comment:
It should be independent of this pass right?

Reply:
https://github.com/openai/triton/pull/3097